module/VuFind/src/VuFind/Marc/MarcReader.php

```diff
@@ -277,4 +277,161 @@ public function getFieldsSubfields(string $fieldTag, array $subfieldCodes,
+        string $occurrence = '', ?array $subfieldCodes = null
+    ): array {
+        $results
+            = $this->getLinkedFields($fieldTag, $linkedFieldTag, $subfieldCodes);
+        if ('' === $occurrence) {
+            return $results[0] ?? [];
+        }
+        foreach ($results as $field) {
+            if ($occurrence === $field['link']['occurrence']) {
+                return $field;
+            }
+        }
+        return [];
+    }
+
+    /**
+     * Return an array of associative arrays for a linked field such as 880
+     * (Alternate Graphic Representation)
+     *
+     * @param string $fieldTag       The MARC field that contains the linked fields
+     * @param string $linkedFieldTag The linked MARC field tag to get
+     * @param array  $subfieldCodes  The MARC subfield codes to get, or empty for all
+     *
+     * @return array
+     */
+    public function getLinkedFields(string $fieldTag, string $linkedFieldTag,
+        ?array $subfieldCodes = null
+    ): array {
+        $result = [];
+
+        foreach ($this->fields[$fieldTag] ?? [] as $field) {
+            if (!is_array($field)) {
+                // Control field
+                continue;
+            }
+            $link
+                = $this->parseLinkageField($this->getInternalSubfield($field, '6'));
+            if ($link['field'] !== $linkedFieldTag) {
+                continue;
+            }
+            $subfields = [];
+            foreach ($field['s'] ?? [] as $subfield) {
+                if ($subfieldCodes
+                    && !in_array((string)key($subfield), $subfieldCodes)
+                ) {
+                    continue;
+                }
+                $subfields[] = [
+                    'code' => (string)key($subfield),
+                    'data' => current($subfield),
+                ];
+            }
+            if ($subfields) {
+                $result[] = [
+                    'tag' => $fieldTag,
+                    'i1' => $field['i1'],
+                    'i2' => $field['i2'],
+                    'subfields' => $subfields,
+                    'link' => $link
+                ];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Return an array of all values extracted from the specified linked
+     * field/subfield combination.  If multiple subfields are specified and $concat
+     * is true, they will be concatenated together in the order listed -- each entry
+     * in the array  will correspond with a single MARC field.  If $concat is false,
+     * the return array will contain separate entries for separate subfields.
+     *
+     * @param string $fieldTag       The MARC field that contains the linked fields
+     * @param string $linkedFieldTag The linked MARC field tag to get
+     * @param array  $subfieldCodes  The MARC subfield codes to get
+     * @param bool   $concat         Should we concatenate subfields?
```

Comment:
Do we need separate $concat / $separator parameters, or could we assume that if you provide a separator, you want concatenation? Maybe that's too obtuse, but it seems like this might be more verbose than necessary.

Reply:
No, we don't need it, but it's just been so that it has the same parameters as getFieldsSubfields, which in turn derived its parameters from MarcReaderTrait::getFieldArray. I'll change the interface of MarcReader while keeping the trait intact, ok?

Reply:
Once again, I think I'm offering you advice but I'm actually criticizing past me. ;-) I'll leave it entirely to your discretion whether it's worth simplifying this or easier to leave it consistent with the trait. I agree that whatever you do here, we shouldn't mess with the trait -- at least not without opening a whole separate PR to discuss it, which is unlikely to be worth the effort at this stage!

Reply:
It's ok to revisit your old decisions. :) I've now simplified the MarcReader methods while leaving the trait alone. I also added a couple of tests to cover the different cases and fixed non-concatenation results of getLinkedFieldsSubfields.